src/emulator/main.cpp

```diff
@@ -111,6 +115,14 @@ int main(int argc, char *argv[]) {
             glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
             gui::draw_begin(host);
 
+#ifdef USE_DISCORD_RICH_PRESENCE
+            if (host.cfg.discord_rich_presence) {
+                discord::initialize();
```

Comment:
This isn't a request to change anything yet, just curious. Is it supposed to initialize discord every frame like this? Or just update presence?

Reply:
It's supposed to initialize on every frame because I call `shutdown()` when the condition is false. Maybe you have a better way of doing this?

Reply:
There's many ways but the simplest would be to keep a `discord_rich_presence_old` before the loop and then after the loop compare it with the new one. If it goes from off to on, shutdown and the inverse.
"Idle" would not work in that specific case (unless you do the check inside the loop and then toggle one of the two).

Maybe it's better to make a `discord::update_init_status` with the cfg.discord_rich_presence as an arg or something and  handle that logic in there though to not pollute main.